src/workerd/api/global-scope.c++

```diff
@@ -915,4 +915,56 @@ bool Navigator::sendBeacon(jsg::Lock& js, kj::String url,
   return false;
 }
 
+// ======================================================================================
+
+Immediate::Immediate(IoContext& context, TimeoutId timeoutId)
+    : contextRef(context.getWeakRef()),
+      timeoutId(timeoutId) {}
+
+void Immediate::dispose() {
+  contextRef->runIfAlive([&](IoContext& context) {
+    context.clearTimeoutImpl(timeoutId);
+  });
+}
+
+jsg::Ref<Immediate> ServiceWorkerGlobalScope::setImmediate(
+    jsg::Lock& js,
+    jsg::Function<void(jsg::Arguments<jsg::Value>)> function,
+    jsg::Arguments<jsg::Value> args) {
+
+  // This is an approximation of the Node.js setImmediate global API.
+  // We implement it in terms of setting a 0 ms timeout. This is not
+  // how Node.js does it so there will be some edge cases where the
+  // timing of the callback will differ relative to the equivalent
+  // operations in Node.js. For the vast majority of cases, users
+  // really shouldn't be able to tell a difference. It would likely
+  // only be somewhat pathological edge cases that could be affected
+  // by the differences. Unfortunately, changing this later to match
+  // Node.js would likely be a breaking change for some users that
+  // would require a compat flag... but that's ok for now?
+
+  auto& context = IoContext::current();
+  function.setReceiver(js.v8Ref<v8::Value>(js.v8Context()->Global()));
```

Comment:
I believe this line is redundant. Since the `function` was not obtained from a struct, JSG will already have set its receiver to the global object.